lab_4_retrieval_w_clustering/main.py

```diff
@@ -67,6 +81,13 @@ def vectorize(self, tokenized_document: list[str]) -> Vector:
         Returns:
             Vector: BM25 vector for document.
         """
+        if not tokenized_document or not isinstance(tokenized_document, list) \
+            or not all(isinstance(token, str) for token in tokenized_document):
+            raise ValueError
+        if self._calculate_bm25(tokenized_document) is None:
```

Comment:
You call the same method twice. Why?

Reply:
First for None check, separately - not to create a variable

Reply:
Really? I clearly understand it. But I suggest saving the result to variable and call this method once

Reply:
Done